testing/conformance/src/vm.rs

```diff
@@ -234,137 +229,85 @@ where
-        // This restricts the ability for the TestKernel to itself be wrapped by another kernel type.
-        self.0
-            .send::<Self>(recipient, method, params, value, gas_limit, flags)
-    }
-
-    fn upgrade_actor<KK>(&mut self, new_code_cid: Cid, params_id: BlockId) -> Result<CallResult> {
-        self.0.upgrade_actor::<Self>(new_code_cid, params_id)
-    }
-}
-
-impl<M, C, K> SyscallHandler<TestKernel<K>> for TestKernel<K>
-where
-    M: Machine,
-    C: CallManager<Machine = TestMachine<M>>,
-    K: Kernel<CallManager = C>,
-{
-    fn bind_syscalls(
-        &self,
-        _linker: &mut Linker<InvocationData<TestKernel<K>>>,
-    ) -> anyhow::Result<()> {
-        Ok(())
-    }
-}
-
-impl<M, C, K> ActorOps for TestKernel<K>
-where
-    M: Machine,
-    C: CallManager<Machine = TestMachine<M>>,
-    K: Kernel<CallManager = C>,
-{
-    fn resolve_address(&self, address: &Address) -> Result<ActorID> {
-        self.0.resolve_address(address)
-    }
-
-    fn get_actor_code_cid(&self, id: ActorID) -> Result<Cid> {
-        self.0.get_actor_code_cid(id)
-    }
-
-    fn next_actor_address(&self) -> Result<Address> {
-        self.0.next_actor_address()
+    fn into_inner(self) -> (Self::CallManager, BlockRegistry)
+    where
+        Self: Sized,
+    {
+        self.0.into_inner()
     }
 
-    fn create_actor(
-        &mut self,
-        code_id: Cid,
-        actor_id: ActorID,
-        delegated_address: Option<Address>,
-    ) -> Result<()> {
-        self.0.create_actor(code_id, actor_id, delegated_address)
+    fn machine(&self) -> &<Self::CallManager as CallManager>::Machine {
+        self.0.machine()
     }
 
-    fn install_actor(&mut self, _code_id: Cid) -> Result<()> {
-        Ok(())
+    fn limiter_mut(&mut self) -> &mut Self::Limiter {
+        self.0.limiter_mut()
     }
 
-    fn get_builtin_actor_type(&self, code_cid: &Cid) -> Result<u32> {
-        self.0.get_builtin_actor_type(code_cid)
+    fn price_list(&self) -> &PriceList {
+        self.0.price_list()
     }
 
-    fn get_code_cid_for_type(&self, typ: u32) -> Result<Cid> {
-        self.0.get_code_cid_for_type(typ)
+    fn charge_gas(&self, name: &str, compute: Gas) -> Result<GasTimer> {
+        self.0.charge_gas(name, compute)
     }
 
-    fn balance_of(&self, actor_id: ActorID) -> Result<TokenAmount> {
-        self.0.balance_of(actor_id)
+    fn gas_available(&self) -> Gas {
+        self.0.gas_available()
     }
 
-    fn lookup_delegated_address(&self, actor_id: ActorID) -> Result<Option<Address>> {
-        self.0.lookup_delegated_address(actor_id)
+    fn gas_used(&self) -> Gas {
+        self.0.gas_available()
     }
 }
 
-impl<M, C, K> IpldBlockOps for TestKernel<K>
+impl<M, C, K> CallOps<Self> for TestKernel<K>
 where
     M: Machine,
     C: CallManager<Machine = TestMachine<M>>,
-    K: Kernel<CallManager = C>,
+    K: Kernel<CallManager = C> + CallOps<Self>,
```

Comment:
So, nothing worked except to make this code non-generic. Honestly, there's no reason for the test kernel to be generic over the kernel, so that's probably fine?